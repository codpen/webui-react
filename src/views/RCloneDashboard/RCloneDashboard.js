import React from 'react';
import {Card, CardBody, CardHeader, Row} from "reactstrap";
import axiosInstance from "../../utils/API/API";


function RCloneVersion({data, hasError}) {
    if (hasError) {
        return (<p>Error loading.</p>);
    }
    return (
        <Card sm={12} lg={4} md={6}>
            <CardHeader>Version</CardHeader>
            <CardBody>
                <p><strong>Arch:</strong>{data.arch}</p>
                <p><strong>goVersion:</strong>{data.goVersion}</p>
                <p><strong>OS:</strong>{data.os}</p>
                <p><strong>Rclone version:</strong>{data.version}</p>
                <p><strong>isGit:</strong>{data.isGit}</p>
            </CardBody>
        </Card>
    )
}

function OptionsView({data, hasError}) {
    if (hasError) {
        return (<p>Error loading.</p>);
    }
    return (
        <Card sm={12} lg={4} md={6}>
            <CardHeader>Options</CardHeader>
            <CardBody>
            </CardBody>
        </Card>
    )


}

class RCloneDashboard extends React.Component {

    getRcloneStatus = () => {
        axiosInstance.post("core/version").then((res) => {
            this.setState({
                version: res.data,
                hasError: false
            })
        }, () => {
            this.setState({
                hasError: true
            })
        })
    };
    getMemStats = () => {
        axiosInstance.post("core/memstats").then((res) => {
            this.setState({
                memStats: res.data
            })
        })
    };
    getOptions = () => {
        axiosInstance.post("options/get").then((res) => {
            this.setState({
                options: res.data
            })
        })
    };

    componentWillMount() {
        this.getRcloneStatus();
    }

    constructor(props, context) {
        super(props, context);
        this.state = {
            version: {},
            hasError: false,
            memStats: {},
            options: {}
        }
    }


    render() {
        return (
            <div data-test="backendComponent">
                <Row>
                    <RCloneVersion data={this.state.version} hasError={this.state.hasError}/>
                    <OptionsView data={this.state.options} hasError={this.state.hasError}/>
                </Row>
            </div>);
    }
}

// const mapStateToProps = state => ({
//
// });

export default RCloneDashboard;
