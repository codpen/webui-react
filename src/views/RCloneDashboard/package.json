{
  "name": "Rclone Dashboard",
  "version": "0.0.0",
  "private": true,
  "main": "RCloneDashboard.js"
}
