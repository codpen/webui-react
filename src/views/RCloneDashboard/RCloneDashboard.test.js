import React from "react";
import {shallow} from "enzyme";
import {findByTestAtrr, testStore} from "../../../Utils";
import toJson from "enzyme-to-json";
import RCloneDashboard from "./RCloneDashboard";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    return shallow(<RCloneDashboard {...props} store={store}/>);
};


describe('RClone Backend Component', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {};
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAtrr(wrapper, "backendComponent");
            expect(component).toHaveLength(1);
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });
    });

});