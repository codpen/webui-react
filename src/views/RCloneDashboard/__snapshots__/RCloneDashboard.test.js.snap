// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RClone Backend Component renders should match snapshot 1`] = `
<div
  data-test="backendComponent"
>
  <Row
    tag="div"
  >
    <RCloneVersion
      data={Object {}}
      hasError={false}
    />
  </Row>
</div>
`;
